homeassistant/components/config/config_entries.py

```diff
@@ -153,8 +155,19 @@ async def get(self, request: web.Request) -> NoReturn:
     @require_admin(
         error=Unauthorized(perm_category=CAT_CONFIG_ENTRIES, permission="add")
     )
-    async def post(self, request: web.Request) -> web.Response:
+    @RequestDataValidator(
+        vol.Schema(
+            {
+                vol.Required("handler"): vol.Any(str, list),
+                vol.Optional("show_advanced_options", default=False): cv.boolean,
+                vol.Optional("entry_id"): cv.string,
+            },
+            extra=vol.ALLOW_EXTRA,
+        )
+    )
+    async def post(self, request: web.Request, data: dict[str, Any]) -> web.Response:
         """Handle a POST request."""
+        self._data = data
         try:
             return await super().post(request)
         except DependencyError as exc:
```

Comment:
This is not OK, because there's a single instance of the `ConfigManagerFlowIndexView` class handling all requests. We should instead split up `FlowManagerIndexView.post` in an outer function, which is decorated with `@RequestDataValidator`, and an inner function, `FlowManagerIndexView._post_impl` or something like that, which child classes which have their own `@RequestDataValidator` can call.